lib/clouds/libcloud_common.py

```diff
@@ -108,29 +122,37 @@ class LibcloudCmds(CommonCloudFunctions) :
 
     @trace
     def __init__ (self, pid, osci, expid = None, provider = "OverrideMe", \
-                  num_credentials = 2, use_ssh_keys = False, \
-                  use_cloud_init = False,  use_volumes = False, use_locations = True, \
-                  use_sizes = True, tldomain = False, verify_ssl = True, extra = {}) :
+                  num_credentials = 2, use_ssh_keys = False, use_volumes = False, \
+                  use_locations = True, use_services = False, use_networks = False, \
+                  use_security_groups = False, use_floating_ips = False, \
+                  use_public_ips = True, use_sizes = True, use_get_image = True, \
+                  tldomain = False, verify_ssl = True, extra = {}) :
         '''
         TBD
         '''
         CommonCloudFunctions.__init__(self, pid, osci, expid)
-        self.access_url = False
         self.ft_supported = False
         self.current_token = 0
         self.cache_mutex = threading.Lock()
         self.token_mutex = threading.Lock()
         self.provider = provider
         self.num_credentials = num_credentials
         self.use_ssh_keys = use_ssh_keys
-        self.use_cloud_init = use_cloud_init
         self.use_volumes = use_volumes
         self.use_locations = use_locations
+        self.use_services = use_services
+        self.use_networks = use_networks
+        self.use_security_groups = use_security_groups
+        self.use_floating_ips = use_floating_ips
+        self.use_public_ips = use_public_ips        
         self.use_sizes = use_sizes
+        self.use_get_image = use_get_image
```

Comment:
As we were discussing on mattermost, we really have to stop the key/value bleeding. I'm totally fine with REMOVING all of these use_xxxx statements that I introduced, so long as we find a way to do this in a configuration file templates that is shared amongst all the libcloud-based adapters without requiring the user to have to define them all or even know about them all in advance to get started developing a new cloud adapter.

Reply:
I am ok on removing these too, over time.